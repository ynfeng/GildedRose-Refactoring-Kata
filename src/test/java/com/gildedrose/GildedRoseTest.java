package com.gildedrose;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void should_align_with_base_line() {
        CombinationApprovals.verifyAllCombinations(GildedRoseTest::itemOneDayPassed,
            new String[] {
                "foo",
                "Aged Brie",
                "Conjured",
                "Backstage passes to a TAFKAL80ETC concert",
                "Sulfuras, Hand of Ragnaros"},
            new Integer[] {0, -1, 4, 12, 9},
            new Integer[] {0, 3, 52, 47});
    }

    @Test
    public void conjured_should_decrease_quality_twice_than_of_ordinary_item() {
        Item item = ItemFactory.createItem("Conjured", 10, 15);

        item.oneDayPassed();

        assertEquals(item.sellIn(), 9);
        assertEquals(item.quality(), 13);
    }

    @Test
    public void conjured_should_decrease_quality_twice_than_of_ordinary_item_when_expire() {
        Item item = ItemFactory.createItem("Conjured", 0, 15);

        item.oneDayPassed();

        assertTrue(item.isExpired());
        assertEquals(item.quality(), 11);
    }

    private static String itemOneDayPassed(String name, int sellIn, int quality) {
        Item[] items = {ItemFactory.createItem(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.oneDayPassed();
        return app.items()[0].toString();
    }

}
