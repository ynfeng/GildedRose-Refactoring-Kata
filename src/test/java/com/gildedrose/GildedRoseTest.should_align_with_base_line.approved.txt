[foo, 0, 0] => foo, -1, 0 
[foo, 0, 3] => foo, -1, 1 
[foo, 0, 52] => foo, -1, 50 
[foo, 0, 47] => foo, -1, 45 
[foo, -1, 0] => foo, -2, 0 
[foo, -1, 3] => foo, -2, 1 
[foo, -1, 52] => foo, -2, 50 
[foo, -1, 47] => foo, -2, 45 
[foo, 4, 0] => foo, 3, 0 
[foo, 4, 3] => foo, 3, 2 
[foo, 4, 52] => foo, 3, 51 
[foo, 4, 47] => foo, 3, 46 
[foo, 12, 0] => foo, 11, 0 
[foo, 12, 3] => foo, 11, 2 
[foo, 12, 52] => foo, 11, 51 
[foo, 12, 47] => foo, 11, 46 
[foo, 9, 0] => foo, 8, 0 
[foo, 9, 3] => foo, 8, 2 
[foo, 9, 52] => foo, 8, 51 
[foo, 9, 47] => foo, 8, 46 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 3] => Aged Brie, -1, 5 
[Aged Brie, 0, 52] => Aged Brie, -1, 52 
[Aged Brie, 0, 47] => Aged Brie, -1, 49 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 3] => Aged Brie, -2, 5 
[Aged Brie, -1, 52] => Aged Brie, -2, 52 
[Aged Brie, -1, 47] => Aged Brie, -2, 49 
[Aged Brie, 4, 0] => Aged Brie, 3, 1 
[Aged Brie, 4, 3] => Aged Brie, 3, 4 
[Aged Brie, 4, 52] => Aged Brie, 3, 52 
[Aged Brie, 4, 47] => Aged Brie, 3, 48 
[Aged Brie, 12, 0] => Aged Brie, 11, 1 
[Aged Brie, 12, 3] => Aged Brie, 11, 4 
[Aged Brie, 12, 52] => Aged Brie, 11, 52 
[Aged Brie, 12, 47] => Aged Brie, 11, 48 
[Aged Brie, 9, 0] => Aged Brie, 8, 1 
[Aged Brie, 9, 3] => Aged Brie, 8, 4 
[Aged Brie, 9, 52] => Aged Brie, 8, 52 
[Aged Brie, 9, 47] => Aged Brie, 8, 48 
[Conjured, 0, 0] => Conjured, -1, 0 
[Conjured, 0, 3] => Conjured, -1, 0 
[Conjured, 0, 52] => Conjured, -1, 48 
[Conjured, 0, 47] => Conjured, -1, 43 
[Conjured, -1, 0] => Conjured, -2, 0 
[Conjured, -1, 3] => Conjured, -2, 0 
[Conjured, -1, 52] => Conjured, -2, 48 
[Conjured, -1, 47] => Conjured, -2, 43 
[Conjured, 4, 0] => Conjured, 3, 0 
[Conjured, 4, 3] => Conjured, 3, 1 
[Conjured, 4, 52] => Conjured, 3, 50 
[Conjured, 4, 47] => Conjured, 3, 45 
[Conjured, 12, 0] => Conjured, 11, 0 
[Conjured, 12, 3] => Conjured, 11, 1 
[Conjured, 12, 52] => Conjured, 11, 50 
[Conjured, 12, 47] => Conjured, 11, 45 
[Conjured, 9, 0] => Conjured, 8, 0 
[Conjured, 9, 3] => Conjured, 8, 1 
[Conjured, 9, 52] => Conjured, 8, 50 
[Conjured, 9, 47] => Conjured, 8, 45 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 3] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 52] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 47] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 3] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 52] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 47] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 4, 0] => Backstage passes to a TAFKAL80ETC concert, 3, 3 
[Backstage passes to a TAFKAL80ETC concert, 4, 3] => Backstage passes to a TAFKAL80ETC concert, 3, 6 
[Backstage passes to a TAFKAL80ETC concert, 4, 52] => Backstage passes to a TAFKAL80ETC concert, 3, 52 
[Backstage passes to a TAFKAL80ETC concert, 4, 47] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 3] => Backstage passes to a TAFKAL80ETC concert, 11, 4 
[Backstage passes to a TAFKAL80ETC concert, 12, 52] => Backstage passes to a TAFKAL80ETC concert, 11, 52 
[Backstage passes to a TAFKAL80ETC concert, 12, 47] => Backstage passes to a TAFKAL80ETC concert, 11, 48 
[Backstage passes to a TAFKAL80ETC concert, 9, 0] => Backstage passes to a TAFKAL80ETC concert, 8, 2 
[Backstage passes to a TAFKAL80ETC concert, 9, 3] => Backstage passes to a TAFKAL80ETC concert, 8, 5 
[Backstage passes to a TAFKAL80ETC concert, 9, 52] => Backstage passes to a TAFKAL80ETC concert, 8, 52 
[Backstage passes to a TAFKAL80ETC concert, 9, 47] => Backstage passes to a TAFKAL80ETC concert, 8, 49 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 3] => Sulfuras, Hand of Ragnaros, 0, 3 
[Sulfuras, Hand of Ragnaros, 0, 52] => Sulfuras, Hand of Ragnaros, 0, 52 
[Sulfuras, Hand of Ragnaros, 0, 47] => Sulfuras, Hand of Ragnaros, 0, 47 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 3] => Sulfuras, Hand of Ragnaros, -1, 3 
[Sulfuras, Hand of Ragnaros, -1, 52] => Sulfuras, Hand of Ragnaros, -1, 52 
[Sulfuras, Hand of Ragnaros, -1, 47] => Sulfuras, Hand of Ragnaros, -1, 47 
[Sulfuras, Hand of Ragnaros, 4, 0] => Sulfuras, Hand of Ragnaros, 4, 0 
[Sulfuras, Hand of Ragnaros, 4, 3] => Sulfuras, Hand of Ragnaros, 4, 3 
[Sulfuras, Hand of Ragnaros, 4, 52] => Sulfuras, Hand of Ragnaros, 4, 52 
[Sulfuras, Hand of Ragnaros, 4, 47] => Sulfuras, Hand of Ragnaros, 4, 47 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 3] => Sulfuras, Hand of Ragnaros, 12, 3 
[Sulfuras, Hand of Ragnaros, 12, 52] => Sulfuras, Hand of Ragnaros, 12, 52 
[Sulfuras, Hand of Ragnaros, 12, 47] => Sulfuras, Hand of Ragnaros, 12, 47 
[Sulfuras, Hand of Ragnaros, 9, 0] => Sulfuras, Hand of Ragnaros, 9, 0 
[Sulfuras, Hand of Ragnaros, 9, 3] => Sulfuras, Hand of Ragnaros, 9, 3 
[Sulfuras, Hand of Ragnaros, 9, 52] => Sulfuras, Hand of Ragnaros, 9, 52 
[Sulfuras, Hand of Ragnaros, 9, 47] => Sulfuras, Hand of Ragnaros, 9, 47 
